prisma/migrations/20250305175301/migration.sql

```diff
@@ -0,0 +1,47 @@
+-- exclude posts in own subs from spam detection
+CREATE OR REPLACE FUNCTION item_spam(parent_id INTEGER, user_id INTEGER, within INTERVAL)
+RETURNS INTEGER
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    repeats INTEGER;
+    self_replies INTEGER;
+BEGIN
+    -- no fee escalation
+    IF within = interval '0' THEN
+        RETURN 0;
+    END IF;
+
+    SELECT count(*) INTO repeats
+    FROM "Item"
+    JOIN "Sub" ON "Sub"."name" = "Item"."subName"
+    WHERE (
+        (parent_id IS NULL AND "parentId" IS NULL)
+        OR
+        ("parentId" = parent_id AND user_id <> (SELECT i."userId" FROM "Item" i WHERE i.id = "Item"."rootId"))
+    )
+    AND "Item"."userId" = user_id
+    AND "bio" = 'f'
+    AND "Sub"."userId" <> user_id
+    AND "Item".created_at > now_utc() - within;
```

Comment:
Comments don't have `subName` set so the `JOIN` will return an empty table. This means that spam fees are no longer applied to replies. You need to use `LEFT JOIN` and handle the reply case:

```suggestion
    LEFT JOIN "Sub" ON "Sub"."name" = "Item"."subName"
    WHERE (
        (parent_id IS NULL AND "parentId" IS NULL)
        OR
        ("parentId" = parent_id AND user_id <> (SELECT i."userId" FROM "Item" i WHERE i.id = "Item"."rootId"))
    )
    AND "Item"."userId" = user_id
    AND "bio" = 'f'
    AND ("Sub"."name" IS NULL OR "Sub"."userId" <> user_id)
    AND "Item".created_at > now_utc() - within;
```

Reply:
Good catch, I didn't think of comments!